### ISSUE TITLE
Multiple keyboard shortcuts support per command

### ISSUE THREAD

**Issue by gerazov** [spyder-ide/spyder#10787]
Moved to Spyder 4 RC2 - awesome work guys! The dark theme is awesome!

I miss something from previous Spyders  though - `run cell` used to work with both the main big `Return`, and the small `Enter` at the corner of the `NumPad`, i.e. both of them mapped to the same `Return` key in previous Spyders. But now they're differentiated.  

It's more convenient to be able to use both shortcuts for the same command. Now I can set both `Ctrl+Return`, and `Ctrl+Enter` for the shortcut, but Spyder interprets this as a sequence.

**Comment by ccordoba12** [comment]
Thanks for noticing this small regression. We'll fix it after we release Spyder 4.

**Comment by jnsebgosselin** [comment]
> 
> Thanks for noticing this small regression. We'll fix it after we release Spyder 4.

I can take care of that one if you wish.

**Comment by goanpeca** [comment]
Please go ahead @jnsebgosselin (side note, I will upload the translations tonight, sorry for the delays!)

**Comment by jnsebgosselin** [comment]
I can think of two options to solve this:

1. We hardcode `Ctrl+Enter` and `Shift+Return` as shortcut for `run cell` and `run cell and advance`.
2. We add the possibility to define 2 shortcuts per action in the `Keyboard shortcuts` preference.

Or maybe there is another option. Any thoughts?

**Comment by goanpeca** [comment]
>2. We add the possibility to define 2 shortcuts per action in the Keyboard shortcuts preference.

We will do this for Spyder 5, which is something we wanted to do for some time now.

>We hardcode Ctrl+Enter and Shift+Return as shortcut for run cell and run cell and advance.

We should probably do this for now to avoid the regression. Thoughts @ccordoba12 ?

**Comment by goanpeca** [comment]
@jnsebgosselin we talked a bit with @ccordoba12. Do you think you could work on this by adding an extra column to the shortcuts section on the preferences pane?

The rationale is that having more than 2 shortcuts/aliases is too much so we can keep the interface simple by adding an extra column. 

We would also need to make the entries on the configuration lists and handle this appropriately.

Thoughts?

**Comment by jnsebgosselin** [comment]
> @jnsebgosselin we talked a bit with @ccordoba12. Do you think you could work on this by adding an extra column to the shortcuts section on the preferences pane?
> 
> The rationale is that having more than 2 shortcuts/aliases is too much so we can keep the interface simple by adding an extra column.
> 
> We would also need to make the entries on the configuration lists and handle this appropriately.
> 
> Thoughts?

Yes, I agree with this, from my experience, having 2 configurable shortcuts per action seems to be the norm.

I will start working on this 👍

**Comment by goanpeca** [comment]
Great! thanks a lot :-)

**Comment by jnsebgosselin** [comment]
@ccordoba12 @goanpeca 
What is the strategy you would like to use to store the key sequences in the configs, now that we can have two key sequence assigned per shortcut?

Would it be ok to store them using a list of strings? This would make it very convenient for assigning shortcuts to `QAction`.

**Comment by ccordoba12** [comment]
> Would it be ok to store them using a list of strings? This would make it very convenient for assigning shortcuts to QAction

That's fine for me and what I was about to suggest.

**Comment by jnsebgosselin** [comment]
Here is a POC where I added an extra column to the shortcuts section on the preferences pane and represented each shortcut by a button. Any thoughts?

![keyboard_pref_wip](https://user-images.githubusercontent.com/10170372/69560271-7e247780-0f79-11ea-9f3a-3db785e890e9.gif)

**Comment by goanpeca** [comment]
@jnsebgosselin that looks awesome !

Is it possible to trigger the shortcut editor with the keyboard also? (instead of mouse click?)

Please open a PR :-)

**Comment by jnsebgosselin** [comment]
> 
> 
> @jnsebgosselin that looks awesome !
> 
> Is it possible to trigger the shortcut editor with the keyboard also? (instead of mouse click?)
> 
> Please open a PR :-)

Yes, it is still possible to trigger the shortcut editor with the keyboard when the table has focus.

However, It was previously possible to navigate the table and trigger the shortcut editor while the search line edit had focus. I'm not sure how to handle that case with this design though. I don't know if it's a big deal, since returning the focus to the table can easily be done with a simple Shift-Tab.

I'll open  PR and we can discuss this over there.

**Comment by goanpeca** [comment]
(imple Shift-Tab. yeah thats ok!)

**Comment by ccordoba12** [comment]
It looks pretty neat, thanks @jnsebgosselin!
